libraries/Microsoft.Bot.Builder.Dialogs/SkillDialog.cs

```diff
@@ -197,5 +208,66 @@ private async Task<Activity> SendToSkillAsync(ITurnContext context, Activity act
 
             return eocActivity;
         }
+
+        private async Task<bool> InterceptOAuthCardsAsync(ITurnContext turnContext, Activity activity, string connectionName, CancellationToken cancellationToken)
+        {
+            if (activity?.Attachments != null && !string.IsNullOrWhiteSpace(connectionName))
+            {
+                foreach (var attachment in activity.Attachments.Where(a => a?.ContentType == OAuthCard.ContentType))
+                {
+                    var oauthCard = ((JObject)attachment.Content).ToObject<OAuthCard>();
+                    if (!string.IsNullOrWhiteSpace(oauthCard.TokenExchangeResource?.Uri))
+                    {
+                        var tokenExchangeProvider = turnContext.Adapter as IExtendedUserTokenProvider;
+                        if (tokenExchangeProvider == null)
+                        {
+                            return false;
+                        }
+
+                        TokenResponse result;
+                        try
+                        {
+                            result = await tokenExchangeProvider.ExchangeTokenAsync(
+                               turnContext,
+                               connectionName,
+                               turnContext.Activity.From.Id,
+                               new TokenExchangeRequest(oauthCard.TokenExchangeResource.Uri)).ConfigureAwait(false);
+                        }
+                        catch
+                        {
+                            return false;
+                        }
+
+                        if (!string.IsNullOrWhiteSpace(result?.Token))
```

Comment:
What happens if token is null?

Reply:
falls back to the last "return false" statement and hence the user is shown an oauth card